File: dotnet/ThirtyFour.UI.Tests/Form/BooleanElementTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using ThirtyFour.UI.Form;

namespace ThirtyFour.UI.Tests.Form
{
    [TestClass]
    public class BooleanElementTests : BaseTestSuite
    {
        public class BooleanElementImpl : BooleanElement
        {
            public BooleanElementImpl(IWebElement element) : base(element)
            {
            }
        }

        [TestMethod]
        public void IsCheckedPropertyWhenNotChecked()
        {
            var el = driver.FindElement(By.Id("rorange"));
            BooleanElement radio = new BooleanElementImpl(el);
            Assert.IsFalse(radio.IsChecked);
        }

        [TestMethod]
        public void IsCheckedPropertyWhenChecked()
        {
            var el = driver.FindElement(By.Id("rviolet"));
            BooleanElement radio = new BooleanElementImpl(el);
            Assert.IsTrue(radio.IsChecked);
        }
    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/form/BooleanElementTests.java
package com.github.danhumphrey.thirtyfour.ui.test.form;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.github.danhumphrey.thirtyfour.ui.form.BooleanElement;
import com.github.danhumphrey.thirtyfour.ui.test.BaseTestSuite;

public class BooleanElementTests extends BaseTestSuite {

	public class BooleanElementImpl extends BooleanElement {
        public BooleanElementImpl(WebElement element) {
       	 super(element);
        }
    }
	
    @Test
    public void testGetCheckedWhenNotChecked() {
    	WebDriver driver = getDriver();
        WebElement el = driver.findElement(By.id("rorange"));
        BooleanElement radio = new BooleanElementImpl(el);
        Assertions.assertFalse(radio.getChecked());
    }

    @Test
    public void testGetCheckedWhenChecked() {
    	WebDriver driver = getDriver();
        WebElement el = driver.findElement(By.id("rviolet"));
        BooleanElement radio = new BooleanElementImpl(el);
        Assertions.assertTrue(radio.getChecked());
    }
       
}
